eng/common/scripts/Verify-Links.ps1

```diff
@@ -245,6 +254,8 @@ while ($pageUrisToCheck.Count -ne 0)
   Write-Host "Found $($linkUris.Count) links on page $pageUri";
   
   foreach ($linkUri in $linkUris) {
+    $linkUri = ReplaceGithubLink $linkUri
```

Comment:
I'm not sure if we want to blindly replace before we check the link or afterwards only if the link fails. I kind of think we should only do the replacement if the original link check fails. That seems like the better option where we don't end up having rewritten links all the time when we don't intend to.

Reply:
Although one other example is if someone removes a file in a PR we probably want to have the validation fail and in that case we would need to check against the sha first. Perhaps we need a little more logic here to handle that case also. Maybe something like:

```
$linkValid = CheckLink $linkUri

$replacedLinkUri = ReplaceGithubLink $linkUri

if ($linkUri -ne $replacedLinkUri) {
  $replacedLinkValid = CheckLink $replacedLinkUri

  if ($linkValid -and !replacedLinkValid) {
    # This is the case of a removing a file and we should add to badlinks
  }
 if (!$linkValid -and $replacedLinkValid) {
   # This is the case of adding a new file so everything is good.
 }
 if (!$linkValid -and !$replacedLinkValid) {
  # This is the case where it is a bad link in both cases and should be added to badlinks. 
 }
```

Reply:
We talked offline and the original code will handle this code.